Renamed Invert to Mode

williamhatcher opened the pull request.

williamhatcher committed:
Renamed Invert to Mode

williamhatcher commented:
Fixes #18

williamhatcher commented:
Should the version be changed to `1.3.1` or `1.4.0`?

Zerthick commented on src/main/java/io/github/zerthick/protectionperms/ProtectionPerms.java:
```diff
@@ -78,13 +78,18 @@ public void onServerInit(GameInitializationEvent event) {
+                PermHandler.getInstance().init(false);
+            } else if (mode.equals("blacklist")){
+                PermHandler.getInstance().init(true);
+            }
```
There should be an else block here in case the user types something other than "whitelist" and "blacklist", we should also probably display some kind of warning as well

williamhatcher replied:
I agree. Should the mode be set to `whitelist` if the mode does not equal one of them?

williamhatcher replied:
I added an else block.
Please see dec5390 for important notes.

Zerthick replied:
Yeah whitelist should be the default

Zerthick commented on src/main/java/io/github/zerthick/protectionperms/ProtectionPerms.java:
```diff
@@ -78,13 +78,18 @@ public void onServerInit(GameInitializationEvent event) {
-
+            CommentedConfigurationNode modeNode = configLoader.load().getNode("mode");
+            if (modeNode.isVirtual()) {
+                logger.warn("Config corrupt! Using defaults (whitelist mode)\nDelete " + instance.getId() + ".conf to fix");
```
We say that we are using the default mode "whitelist" here, but we never actually set it.

williamhatcher replied:
`String node = modeNode.getString("whitelist");` sets the mode to whitelist, or gets the mode set in the config.

Zerthick replied:
Oops you're right, forgot about that default setting

Zerthick commented on src/main/java/io/github/zerthick/protectionperms/ProtectionPerms.java:
```diff
@@ -78,13 +78,18 @@ public void onServerInit(GameInitializationEvent event) {
+            if (modeNode.isVirtual()) {
+                logger.warn("Config corrupt! Using defaults (whitelist mode)\nDelete " + instance.getId() + ".conf to fix");
+            }
+            String mode = modeNode.getString("whitelist");
```
If the modeNode is virtual we should never try to fetch the value from it in the first place

williamhatcher replied:
I could do
```java
if (modeNode.isVirtual()) {
    logger.warn("Config corrupt! Using defaults (whitelist mode)\nDelete " + instance.getId() + ".conf to fix");
    String mode = "whitelist";
} else {
    String mode = modeNode.getString("whitelist");
}
```

But since `getString()` allows a default value, I use the submitted method because it's simpler. (From a Pythonistic view)

Zerthick replied:
You're right

Zerthick commented on src/main/java/io/github/zerthick/protectionperms/ProtectionPerms.java:
```diff
@@ -78,13 +78,18 @@ public void onServerInit(GameInitializationEvent event) {
+                PermHandler.getInstance().init(true);
+            }
         } catch (IOException e) {
             logger.warn("Error loading config! Error: " + e.getMessage());
```
We should change this from logger.warn to logger.error

williamhatcher replied:
Done.

Zerthick commented:
We should bump the version to 1.4.0, as the actual interface has changed, it's not just a bugfix.

williamhatcher committed:
Changed logger.warn to logger.error

williamhatcher committed:
Added else block to mode detection

The plugin will log a warning if `mode` is not `whitelist` or `blacklist` in the config.

Note: I edited this directly on GitHub because I uninstalled my Java IDE. Please check it.

williamhatcher committed:
Bumped version to 1.4.0

williamhatcher commented:
Before merging this PR, the README has to be updated.

williamhatcher committed:
Update README.md

williamhatcher committed:
Added whitespace to fix formatting.

williamhatcher committed:
Added whitespace to fix formatting.

Zerthick reviewed (approved):
Looks good
